source/server/http/admin.cc

```diff
@@ -653,7 +663,11 @@ void AdminFilter::onComplete() {
   Buffer::OwnedImpl response;
   Http::HeaderMapPtr header_map{new Http::HeaderMapImpl};
   RELEASE_ASSERT(request_headers_);
-  Http::Code code = parent_.runCallback(path, *request_headers_, *header_map, response);
+
+  std::unique_ptr<AdminStreamImpl> admin_stream =
```

Comment:
My suggestion to make this a unique_ptr rather than instantiate it on-stack only makes sense if you are going to hold the unique_ptr in the AdminStream, moving the ownership of header_map and response along with it.

If you really don't want to do that, you might as well just instantiate the AdminStream on the stack here.  You are not releasing it or std::move()ing it either, so unique_ptr locally here doesn't have much impact.